domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt

```diff
@@ -0,0 +1,289 @@
+        getJSONObject(interactionJson,"customization_args")))
+      .build()
+  }
+
+  // Creates the list of answer group objects from JSON
+  private fun createAnswerGroupsFromJson(answerGroupsJson: JSONArray?,
+                                         interactionId: String): MutableList<AnswerGroup> {
+    val answerGroups = mutableListOf<AnswerGroup>()
+    if(answerGroupsJson == null) {
+      return answerGroups
+    }
+    for (i in 0 until answerGroupsJson.length()) {
+      answerGroups.add(createSingleAnswerGroupFromJson(
+        answerGroupsJson.getJSONObject(i), interactionId))
+    }
+    return answerGroups
+  }
+
+  // Creates a single answer group object from JSON
+  private fun createSingleAnswerGroupFromJson(answerGroupJson: JSONObject,
+                                              interactionId: String): AnswerGroup {
+    return AnswerGroup.newBuilder()
+      .setOutcome(
+        createOutcomeFromJson(answerGroupJson.getJSONObject("outcome")))
+      .setCorrect(
+        answerGroupJson.getBoolean("correct"))
+      .addAllRuleSpecs(
+        createRuleSpecsFromJson(
+          answerGroupJson.getJSONArray("rule_specs"),  interactionId))
+      .build()
+  }
+
+  // Creates an outcome object from JSON
+  private fun createOutcomeFromJson(outcomeJson: JSONObject?): Outcome {
+    if(outcomeJson == null) {
+      return Outcome.getDefaultInstance()
+    }
+    return Outcome.newBuilder()
+      .setDestStateName(outcomeJson.getString("dest"))
+      .addAllFeedback(createFeedbackFromJson(outcomeJson.getJSONArray("feedback")))
+      .build()
+  }
+
+  // Creates a feedback object from JSON
+  private fun createFeedbackFromJson(feedbackJson: JSONArray?): MutableList<SubtitledHtml> {
+    val feedbackList = mutableListOf<SubtitledHtml>();
+    if (feedbackJson == null) {
+      return feedbackList
+    }
+    for (i in 0 until feedbackJson.length()) {
+      feedbackList.add(SubtitledHtml.newBuilder().setHtml(feedbackJson.getString(i)).build())
+    }
+    return feedbackList
+  }
+
+  // Creates the list of rule spec objects from JSON
+  private fun createRuleSpecsFromJson(ruleSpecJson: JSONArray?,
+                                      interactionId: String): MutableList<RuleSpec> {
+    val ruleSpecList = mutableListOf<RuleSpec>()
+    if(ruleSpecJson == null) {
+      return ruleSpecList
+    }
+    for (i in 0 until ruleSpecJson.length()) {
+      ruleSpecList.add(
+        RuleSpec.newBuilder()
+          .setRuleType(
+            ruleSpecJson.getJSONObject(i).getString("rule_type"))
+          .setInput(createInputFromJson(
+            ruleSpecJson.getJSONObject(i).getJSONObject("inputs"),
+            /* keyName= */"x", interactionId))
+          .build())
+    }
+    return ruleSpecList
+  }
+
+  // Creates an input interaction object from JSON
+  private fun createInputFromJson(inputJson: JSONObject?,
+                                              keyName: String,
```

Comment:
Ditto here for spacing & argument wrapping.

Reply:
Done